ideaforce/src/main/java/com/emc/ideaforce/controller/AdminController.java

```diff
@@ -44,9 +53,33 @@ public ModelAndView showAdminPage(Principal principal) {
 
     @RequestMapping(value="/approve/{id}", method = GET)
     @RolesAllowed(Utils.ADMIN_ROLE)
-    public String approveStory(Principal principal, @PathVariable String id) {
+    public String approveStory(@PathVariable String id) {
         commonService.setStoryApproved(id);
-        return "approved";
+        return APPROVE_CHALLENGE;
+    }
+
+    @RequestMapping(value="/addcomments/{id}", method = GET)
+    public ModelAndView commentsForStory(@PathVariable String id, ModelMap model) {
+        CommentDto commentDto = new CommentDto();
+        commentDto.setStoryId(id);
+        model.addAttribute("newcomment", commentDto);
+        return new ModelAndView(ADD_COMMENTS_VIEW, model);
+    }
+
+    @RequestMapping(value="/addcomments", method = POST)
+    public ModelAndView addCommentsForStory(@ModelAttribute("newcomment") CommentDto commentModel,
+                                            Principal principal) {
+        User currentUser = userService.getUser(principal.getName());
+        commonService.saveStoryComment(commentModel, currentUser);
+        return showAdminPage();
+    }
+
+    @RequestMapping(value="/viewcomments/{id}", method = GET)
+    public ModelAndView viewCommentsForStory(@PathVariable String id) {
```

Comment:
This api is also required for for normal_user, right? User needs to view the comments given for his story..

Reply:
Can it be part of profile implementation ?

Reply:
Sure